clustered/integration-test/src/test/java/org/ehcache/clustered/TerminatedServerTest.java

```diff
@@ -585,9 +595,58 @@ Void runTask() {
       }.run();
   }
 
+  /**
+   * If the server goes down, the client should not freeze on a server call. It should timeout and answer using
+   * the resilience strategy. Whatever the number of calls is done afterwards.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testTerminationFreezesTheClient() throws Exception {
+    Duration readOperationTimeout = Duration.ofMillis(100);
+
+    try(PersistentCacheManager cacheManager =
+          CacheManagerBuilder.newCacheManagerBuilder()
+            .with(ClusteringServiceConfigurationBuilder.cluster(cluster.getConnectionURI().resolve("/MyCacheManagerName"))
+              .timeouts(TimeoutsBuilder.timeouts()
+                .read(readOperationTimeout))
+              .autoCreate()
+              .defaultServerResource("primary-server-resource"))
+            .withCache("simple-cache",
+              CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class,
+                ResourcePoolsBuilder.newResourcePoolsBuilder()
+                  .with(ClusteredResourcePoolBuilder.clusteredDedicated(4, MemoryUnit.MB))))
+            .build(true)) {
+
+      Cache<Long, String> cache = cacheManager.getCache("simple-cache", Long.class, String.class);
+      cache.put(1L, "un");
+
+      cluster.getClusterControl().terminateAllServers();
+
+      // Fill the inflight queue and check that we wait no longer than the read timeout
+      for (int i = 0; i < CLIENT_MAX_PENDING_REQUESTS; i++) {
+        cache.get(1L);
+      }
+
+      // The resilience strategy will pick it up and not exception is thrown
+      new TimeLimitedTask<Void>(readOperationTimeout.toMillis() * 2, TimeUnit.MILLISECONDS) { // I multiply by 2 to let some room after the expected timeout
+        @Override
+        Void runTask() {
+          cache.get(1L); // the call that could block
+          return null;
+        }
+      }.run();
+
+      // FIXME: It freezes on the close() called by the try-with-resources if we never restart the server.
+      cluster.getClusterControl().startOneServer();
+    } catch(StateTransitionException e) {
+      // FIXME: swallow for now because we wait for #2203 to cleanly reconnect
```

Comment:
Issue #2203 is merged. Does this work now?

Reply:
Yes. I need to try. I've been lazy and wanted to do it in another PR. But I will do it now.

Reply:
Actually, some other tests should now pass as well. But the behaviour seems different depending on the lifecycle method (createCache, destroyCache, removeCache). So I'm digging a bit.

Reply:
I know @AbfrmBlr still has work to do on the destroy path... so that might explain some of that.  If you want to split out in to a separate PR that is fine.